src/main.rs

```diff
@@ -172,10 +205,119 @@ impl ReStreamer {
+            match event {
+                InputEvent::WacomEvent { event: e } =>
+                    match e {
+                        WacomEvent::InstrumentChange { pen: WacomPen::ToolPen, state: s } => {
+                            down = s;
+                        },
+                        WacomEvent::Hover { position: cgmath::Point2 { x, y }, .. } => {
+                            pos = (x as usize, y as usize);
+                            self.drawing = false;
+                        },
+                        WacomEvent::Draw { .. } => {
+                            // no need to show position while drawing
+                            self.drawing = true;
+                        },
+                        _ => (),
+                    },
+                    _ => (),
+            }
+        }
+
+        self.pen_pos = if down {
+            Some(pos)
+        } else {
+            None
+        };
+    }
+
+    /// Draw pen position into fb data, if necessary (in hover range and not drawing).
+    fn draw_pen_position(&mut self, buf: &mut [u8]) {
+        if let (false, Some((y, x))) = (self.drawing, self.pen_pos) {
+            // we need negative numbers to calculate offsets correctly
+            let width = self.width as isize;
+            let height = self.height as isize;
+            let bpp = self.bytes_per_pixel as isize;
+            let cursor = self.cursor as isize;
+            for (i, (yoff, no)) in PEN_IMAGE.iter().enumerate() {
+                // we draw vertically (lines along y)
+                let xoff = i as isize - (PEN_IMAGE.len() as isize / 2);
+                let xstart = x as isize + xoff;
+                // line outside of canvas?
+                if xstart < 0 || xstart >= width {
+                    continue
+                }
+                let mut ystart = (height - y as isize) + yoff;
+                let mut no = *no;
+                // cut-off at sides
+                if ystart < 0 {
+                    no += ystart;
+                    ystart = 0;
+                }
+                if ystart + no > height {
+                    no = height - ystart;
+                }
+                if no <= 0 {
+                    continue
+                }
+                // translate to buf indexes, check bounds and draw
+                let mut px_start = (xstart * height + ystart) * bpp;
+                let mut px_end = px_start + no * bpp;
+                // outside current buf?
+                if px_end < cursor || px_start >= cursor + buf.len() as isize {
+                    continue
+                }
+                // truncate if partially outside
+                if px_start < cursor {
+                    px_start = cursor;
+                }
+                if px_end > cursor + buf.len() as isize {
+                    px_end = cursor + buf.len() as isize;
+                }
+                // invert pixel (on RM2)
+                // TODO: Do something sensible on RM1
```

Comment:
I don't know how the RM1 framebuffer works, inverting each byte like this might be wrong.